### PR TITLE
feat: Ask user about messaging in Degraded conversation (WPB-1771)

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

Users need to be notified that the conversation where he/she going to send a message is degraded. 
For that:
- Added `should_inform_about_verification_before_messaging` flag  into `Conversation` DB table
- added UseCases to observe and update that flag
- added DB trigger to automatically update that flag when the conversation's verification degraded

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 1]
```diff
@@ -1056,6 +1061,19 @@ internal class ConversationDataSource internal constructor(
             }
         }
 
+    override suspend fun updateInformAboutVerificationBeforeMessagingFlag(
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:1064]:
```suggestion
    override suspend fun setInformAboutVerificationBeforeMessagingFlag(
```

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt [hunk 2]
```diff
@@ -1056,6 +1061,19 @@ internal class ConversationDataSource internal constructor(
...
+        conversationId: QualifiedID,
+        updateFlag: Boolean
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt:1066]: suggestion: updatedValue, or even value.

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq [hunk 3]
```diff
@@ -42,7 +42,8 @@ CREATE TABLE Conversation (
     archived_date_time INTEGER AS Instant,
     -- mls_verification_status
     verification_status TEXT AS ConversationEntity.VerificationStatus NOT NULL DEFAULT "NOT_VERIFIED",
-    proteus_verification_status TEXT AS ConversationEntity.VerificationStatus NOT NULL DEFAULT "NOT_VERIFIED"
+    proteus_verification_status TEXT AS ConversationEntity.VerificationStatus NOT NULL DEFAULT "NOT_VERIFIED",
+    should_inform_about_verification_before_messaging INTEGER AS Boolean NOT NULL DEFAULT 0
```
Comment (mchenani) [inline persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq:46]:
```suggestion
    degraded_conversation_notified INTEGER AS Boolean NOT NULL DEFAULT **0**
``` or 
```suggestion
    notified_degraded_conversation INTEGER AS Boolean NOT NULL DEFAULT 0
```

**Review by mchenani — APPROVED** [review]
Looks good👍, added some naming suggestions.

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**